MediaBrowser.Model/Dlna/StreamBuilder.cs

```diff
@@ -1041,159 +1121,56 @@ private static int GetMaxAudioBitrateForTotalBitrate(long totalBitrate)
-                        {
-                            conditions.Add(c);
-                        }
-                    }
-                }
-            }
-
-            foreach (ProfileCondition i in conditions)
+            // Check audio conditions
+            var failedVideoAudioCodecConditions = Enumerable.Empty<ProfileCondition>();
+            if (audioStream != null)
             {
-                if (!ConditionProcessor.IsVideoConditionSatisfied(i, width, height, bitDepth, videoBitrate, videoProfile, videoLevel, videoFramerate, packetLength, timestamp, isAnamorphic, isInterlaced, refFrames, numVideoStreams, numAudioStreams, videoCodecTag, isAvc))
-                {
-                    LogConditionFailure(profile, "VideoCodecProfile", i, mediaSource);
-
-                    var transcodeReason = GetTranscodeReasonForFailedCondition(i);
-                    var transcodeReasons = transcodeReason.HasValue
-                        ? new List<TranscodeReason> { transcodeReason.Value }
-                        : new List<TranscodeReason>();
-
-                    return (null, transcodeReasons);
-                }
+                var audioCodecConditions = GetConditionsForCodec(videoStream!.Codec, mediaSource.Container, profile.CodecProfiles, CodecType.VideoAudio, mediaSource, audioStream);
+                failedVideoAudioCodecConditions = FindFailedConditions(audioCodecConditions, CodecType.VideoAudio, mediaSource, audioStream);
             }
 
-            if (audioStream != null)
+            if (failedContainerConditions.Any() || failedVideoCodecConditions.Any() || failedVideoAudioCodecConditions.Any())
             {
-                string audioCodec = audioStream.Codec;
-                conditions = new List<ProfileCondition>();
-                bool? isSecondaryAudio = mediaSource.IsSecondaryAudio(audioStream);
+                var transcodeReasons = new HashSet<TranscodeReason>();
 
-                foreach (var i in profile.CodecProfiles)
-                {
-                    if (i.Type == CodecType.VideoAudio && i.ContainsAnyCodec(audioCodec, container))
-                    {
-                        bool applyConditions = true;
-                        foreach (ProfileCondition applyCondition in i.ApplyConditions)
-                        {
-                            if (!ConditionProcessor.IsVideoAudioConditionSatisfied(applyCondition, audioChannels, audioBitrate, audioSampleRate, audioBitDepth, audioProfile, isSecondaryAudio))
-                            {
-                                // LogConditionFailure(profile, "VideoAudioCodecProfile.ApplyConditions", applyCondition, mediaSource);
-                                applyConditions = false;
-                                break;
-                            }
-                        }
+                transcodeReasons.UnionWith(GetAllTranscodeReasonsForFailedConditions(failedContainerConditions, profile, "VideoContainerProfile", mediaSource));
```

Comment:
I think the streambuilder is actually new'd everytime it's used, so `transcodeReasons` could just be a private field.